app/model/application.js

```diff
@@ -1,241 +1,445 @@
+// Returns a number on success (-1 on not found)
+ApplicationSchema.methods.releaseFindOneIndex = function(query) {
+  const application = this;
+
+  const i = _.findIndex(application.releases, function(iRelease) {
+    return _.isMatch(iRelease, query)
+  });
+
+  return i;
+}
+
+// Updates or creates a release based on query, updates with object
+// Returns promise of new application object
+ApplicationSchema.methods.releaseUpdateOrCreate = function(query, object) {
+  const application = this;
+  const i = application.releaseFindOneIndex(query);
+
+  if (i === -1) { // Create a new release
+    Application.update({_id: application._id}, {
+      $push: {
+        releases: object,
+      },
+    }, (error, info) => {
+      if (error) {
+        return Promise.reject(error)
       }
+
+      return Application.findOne({_id: application._id});
+    });
+  }
+
+  Application.update({ // Update already created release
+    _id: application._id,
+    releases: query,
+  }, {
+    $set: {
+      'releases.$': object,
+    },
+  }, (error, info) => {
+    if (error) {
+      return Promise.reject(error)
     }
-    if (newReleases && application.state.standby) {
-      application.status = 'NEW RELEASE';
+
+    return Application.findOne({_id: application._id});
+  });
+}
+
+// Fetch all releases from Github
+// Returns saved application on success
+ApplicationSchema.methods.releaseFetchAll = function() {
+  const application = this;
+  const fullName = application.github.fullName;
+  app.log.silly(`application releaseFetchAll called for ${fullName}`);
+
+  const githubReleases =  gh.request(`GET /repos/${fullName}/releases`, {
+    token: application.github.APItoken,
+  })
+  .catch(error => {
+    return Promise.reject(`Received ${error.status} from github`);
+  });
+
+  return Promise.filter(githubReleases, release => {
+    return semver.valid(release.tag_name, true);
+  })
+  .map(release => {
+    return application.releaseUpdateOrCreate({
+      github: {
+        id:      release.id,
+      },
+    }, {
+      github: {
+        id:      release.id,
+        author:  release.author.login,
+        date:    release.published_at,
+        tag:     release.tag_name,
+      },
+      changelog: release.body.match(/.+/g),
+    });
+  })
+  .catch(error => {
+    return Promise.reject(error);
+  })
+  .then(applications => {
+    return Application.findOne({_id: application._id});
+  });
+}
+
+// Application methods
+// Helper function for application status
+// Returns a boolean always
+ApplicationSchema.methods.isStatus = function(query) {
+  return this.status === query;
+}
+
+// Helper function for application problems
+// Returns a boolean always
+ApplicationSchema.methods.hasProblem = function(query) {
+  return _.indexOf(this.problems, query) >= 0;
+}
+
+// Sets an application problem based on status or toggles if no status
+// Returns nothing
+ApplicationSchema.methods.setProblem = function(query, status = !this.hasProblem(query)) {
```

Comment:
I would however recommend renaming `status` to something more descriptive (and ideally explicitly boolean-sounding), like `isSolved`.